### DISCUSSION TITLE
[Idea] Are child themes dead?

### DISCUSSION THREAD

**Discussion by musicpanda** [PrestaShop/PrestaShop#33487] [Ideas]
### The problem

Recently I got for the first time serious about making a child theme that was more than basic.  It was a road of astonishment and disappointment. Prestashop really needs to invest more to make this a viable technique:

 - after I made a child theme the themes page (/admin/index.php/improve/design/themes/) kept showing the preview and title of the parent for the child. No matter how much I changed the preview file or the child's theme.yml, the child looked exactly like its parent. Things changed only when I deleted the /config/themes/childtheme page with its .json file.

 - next problem was that some modules didn't work. The module maker told me that I needed to copy the assets directory of the parent to the child. Nothing in the documentation told me about this. It suggests that child theme implementation is incomplete.

 - I found the manual extremely poor. It tells you to copy the parent's theme.yml to the child. But its description what you need to do next basically ends with the line 'use_parent_assets:'.  It remains a mystery what you should do with the following lines.

### The solution

Have a good look at it.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
They're definitely still alive, I use them a lot, and I agree, there's **a lot** that needs to be documented. There are also plenty of workarounds because of some limitations, but it is how it is. Right now, the project members can't invest time in child themes as there are many more important subjects to do.

We have some experienced developers in our community who I'm sure could answer some questions.

cc @Oksydan @prestarocket @JBLach and many others :-)

> after I made a child theme the themes page (/admin/index.php/improve/design/themes/) kept showing the preview and title of the parent for the child. No matter how much I changed the preview file or the child's theme.yml, the child looked exactly like its parent. Things changed only when I deleted the /config/themes/childtheme page with its .json file.

Yes! This is one of those things that I personally know about, but there's no perfect way of handling it. There should be a way to clear the theme.yml cache, but it may have some implications on the actual theme look in front office. I'm not sure if we can just delete theme.yml from this directory in the live store, I remember it caused some issues when I tried to do it this way.

> next problem was that some modules didn't work. The module maker told me that I needed to copy the assets directory of the parent to the child. Nothing in the documentation told me about this. It suggests that child theme implementation is incomplete.

True. There were some changes in v8 from @Oksydan regarding this subject, but I think they are undocumented :-(

> I found the manual extremely poor. It tells you to copy the parent's theme.yml to the child. But its description what you need to do next basically ends with the line 'use_parent_assets:'. It remains a mystery what you should do with the following lines.

You're right, but we also have hundreds of features with poor documentation or without documentation at all. We did amazing progress with the docs in the last few years, but there's still a lot of work. I'm not sure if we can handle it alone, without the help of the community.
